### ISSUE TITLE
Conflitc with QControls

### ISSUE THREAD

**Issue by BOBILLIER** [LabVIEW-Open-Source/ui-tools#2]
Hi François
Today I have encountered one issue with your "Create customized control" tool. I have run it on LV2011 and it's ok, but under LV2018 it's broken. It seems there is a class name conflict with your Control.lvclass and the same in QControl tools
I have to try to rename your class, but like it's locked, it's not so easy. Have you any suggestions to solve this?
I send 
![Class conflit with QControl](https://user-images.githubusercontent.com/14000714/112008668-fc6d3b80-8b25-11eb-8989-cdbf996a82be.png)
 a message to QControls support to indicate that I have to find this problem.

**Comment by francois-normandin** [comment]
@BOBILLIER I think you'll need to uninstall QControls first (to unlock the UI Tools class) and then change the class name in UI Tools. Finally, when all is relinked to the new namespace, you can reinstall QControls and wouldn't get a conflict.

Of course, the real way to solve the issue would be to put the UI Tools methods inside a lvlib and provide a permanent fix.

**Comment by francois-normandin** [comment]
@BOBILLIER
I created a 2.0.0 package where namespace has been changed to avoid name collision with QControls
This is not a full release in the sense that I will try to find time to clear out a few quirks and re-release fully through VIPM Community network.

For now, please find the VIP file here: 
https://github.com/LabVIEW-Open-Source/ui-tools-controls-addon/releases/tag/2.0.0

J'espère que ça fonctionnera bien. J'ai testé en LV2020, mais sans QControls... sauf que je ne vois pas comment ça causerait problème!

**Comment by BOBILLIER** [comment]
Bonjour François
Merci beaucoup pour cette modification, je viens de tester et la nouvelle version fonctionne,. Par contre, j’ai maintenant 2 possibilités d’accès à cet outil dans le menu Outils de LV.
Un qui fait référence à la nouvelle version et un second qui fait référence au précédent , et qui lui ne fonctionne toujours pas. Comme la nouvelle version fait référence à l’UI Tools 1.4.1.74
Je ne sais pas quel package désinstaller pour que l’ancien menu disparaisse.

J’avais une autre question plus général. Comment peut-on rajouter des modèles (Button Style) ? J’ai essayé de le faire dans la version précédente en LV11 en rajoutant un dossier pour lui dans le dossier des boutons (Labview Data\Lava\Controls Template) avec dedans un contrôle et les 3 images des différents aspects du bouton. Mais cela ne fonctionne pas. Il apparait bien dans la liste des « Button style », mais pas dans le preview et cela génère des erreurs lorsque j’essaye de le créer.
D’ailleurs, je ne trouve pas le dossier des controls template pour la nouvelle version. Il doit être différent de la version LV11 car mon nouveau Button Style n’apparait même pas dans la liste. Je suppose donc qu’il ne fait pas référence au même dossier.
Merci
Eric


De : Francois Normandin ***@***.***
Envoyé : mardi 23 mars 2021 02:45
À : LabVIEW-Open-Source/ui-tools
Cc : Eric Bobillier; Mention
Objet : Re: [LabVIEW-Open-Source/ui-tools] Conflitc with QControls (#2)


@BOBILLIER<https://github.com/BOBILLIER>
I created a 2.0.0 package where namespace has been changed to avoid name collision with QControls
This is not a full release in the sense that I will try to find time to clear out a few quirks and re-release fully through VIPM Community network.

For now, please find the VIP file here:
https://github.com/LabVIEW-Open-Source/ui-tools-controls-addon/releases/tag/2.0.0

J'espère que ça fonctionnera bien. J'ai testé en LV2020, mais sans QControls... sauf que je ne vois pas comment ça causerait problème!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/LabVIEW-Open-Source/ui-tools/issues/2#issuecomment-804520738>

(Admin note: I remove the direct authorization link that was appended to the email reply. This would have allowed anyone to publish a response in your name. Best practice is to remove the original message if using the reply feature from an email)

**Comment by francois-normandin** [comment]
Hmmm, j'aurais cru que l'ancienne version serait désinstallée en installant la nouvelle. Il y a dû y avoir un changement de nom de la librairie quand j'ai recompilé la librairie de base. J'ai des deadline pas mal serrés aujourd'hui alors pourquoi de chance que je puisse y faire quelque chose immédiatement, mais ça m'aiderait d'avoir une vue depuis VIPM pour me montrer quelles versions de UITools sont installées.

Pour les boutons avec un nouveau modèle, je crois qu'il y a une vieille vidéo sur LAVAG, dans le dépôt de code, à ce sujet. Je devrais en refaire une nouvelle.
En gros, il fait que les nouvelles images (PNG) soient identiques aux états dans le bouton qui se trouve dans le même répertoire. Si une image est associée à l'état "enfoncé", alors l'image que utilisée pour créer cet état doit être identique à celle trouvée dans le répertoire.

Je pourrais fonctionner avec un fichier qui décrit quelle image va avec quel état, mais c'était plus simple de comparer le modèle en extrayant les images déjà contenues et en les remplaçant par les modèles, auxquelles le programme a apposer l'icône.

**Comment by BOBILLIER** [comment]
Voici une copie d'écran de VIPM
![image](https://user-images.githubusercontent.com/14000714/112140075-d69e7000-8bd3-11eb-98f8-8725aa822533.png)
Par contre pour mon contrôle,  j'ai mis à jour le contrôle avec les images du même dossier et cela ne fonctionne toujours pas (pas de preview) . Je te joins un zip avec les éléments, si tu veux tester
[NXG.zip](https://github.com/LabVIEW-Open-Source/ui-tools/files/6189241/NXG.zip)

**Comment by francois-normandin** [comment]
Ça semble venir de la librairie add-on control 1.2.
C'est essentiellement la même que j'ai mis à jour, donc c'est que j'ai effectivement un changement de nom de package. Je vais y voir.

Pour le bouton, je regarde ça en arrivant au bureau. Ça prend normalement quelques minutes à faire...

**Comment by francois-normandin** [comment]
@BOBILLIER 
 Wow, I need to make this simpler... :-)
You need to have one image per state... and the name of the image needs to be "Camel Case"...
[Nxg.zip](https://github.com/LabVIEW-Open-Source/ui-tools/files/6190459/Nxg.zip)

The folder name can be capitalized, but the images and their links need to be following a strict rule.
I'll definitely rework this. I don,t know whey this choice was made 10 years ago... but it is not scalable to my liking!

Proof it now works:
![image](https://user-images.githubusercontent.com/11728548/112165565-3e998a00-8bc5-11eb-93c9-89ed9dc49798.png)

**Comment by francois-normandin** [comment]
Install them under vilib>LAVA folder.

![image](https://user-images.githubusercontent.com/11728548/112165466-2590d900-8bc5-11eb-971c-6a760d76f9f7.png)

**Comment by francois-normandin** [comment]
@BOBILLIER 
Est-ce que j'ai ta permission pour inclure ton template NXG dans le prochain build?

**Comment by BOBILLIER** [comment]
Hi François
Thanks for those precisions and for the control template.
You write “You need to have one image per state” . Are-you sur of that ? Because in your another template GlassWeb, you have only 3 images for 6 states.
The top will be to start from one existing control, extract images and add it as template automatically.   I think it’s perhaps possible with your UI manipulation library.
When I have 5 free minutes , I try to create this tool.
Thanks for your help and all your free tools.
And I give you permission to add NXG in next build , Obviously.
Eric

De : Francois Normandin ***@***.***
Envoyé : mardi 23 mars 2021 15:47
À : LabVIEW-Open-Source/ui-tools
Cc : Eric Bobillier; Mention
Objet : Re: [LabVIEW-Open-Source/ui-tools] Conflitc with QControls (#2)


@BOBILLIER<https://github.com/BOBILLIER>
Wow, I need to make this simpler... :-)
You need to have one image per state... and the name of the image needs to be "Camel Case"...
Nxg.zip<https://github.com/LabVIEW-Open-Source/ui-tools/files/6190459/Nxg.zip>

The folder name can be capitalized, but the images and their links need to be following a strict rule.
I'll definitely rework this. I don,t know whey this choice was made 10 years ago... but it is not scalable to my liking!

Proof it now works:

**Comment by francois-normandin** [comment]
You are right... states and number of images are not supposed to be linked.
In this case, the errors I was seeing were to that effect where there were empty paths.
Anyhow, I'll rework this for the next round. If I release a version 2.0.0 officially, I want it to be an improved version of the control generator!
